eos/effects/adaptivearmorhardener.py

```diff
@@ -2,17 +2,207 @@
+                damagePattern.emAmount,
+                damagePattern.emAmount*fit.ship.getModifiedItemAttr('armorEmDamageResonance'),
+                module.getModifiedItemAttr('armorEmDamageResonance'),
+                module.getModifiedItemAttr('armorEmDamageResonance')])
+
+        damagePattern_tuple.append(['Thermal',
+                fit.ship.getModifiedItemAttr('armorThermalDamageResonance'),
+                damagePattern.thermalAmount,
+                damagePattern.thermalAmount * fit.ship.getModifiedItemAttr('armorThermalDamageResonance'),
+                module.getModifiedItemAttr('armorThermalDamageResonance'),
+                module.getModifiedItemAttr('armorThermalDamageResonance')])
+
+        damagePattern_tuple.append(['Kinetic',
+                fit.ship.getModifiedItemAttr('armorKineticDamageResonance'),
+                damagePattern.kineticAmount,
+                damagePattern.kineticAmount*fit.ship.getModifiedItemAttr('armorKineticDamageResonance'),
+                module.getModifiedItemAttr('armorKineticDamageResonance'),
+                module.getModifiedItemAttr('armorKineticDamageResonance')])
+
+        damagePattern_tuple.append(['Explosive',
+                fit.ship.getModifiedItemAttr('armorExplosiveDamageResonance'),
+                damagePattern.explosiveAmount,
+                damagePattern.explosiveAmount*fit.ship.getModifiedItemAttr('armorExplosiveDamageResonance'),
+                module.getModifiedItemAttr('armorExplosiveDamageResonance'),
+                module.getModifiedItemAttr('armorExplosiveDamageResonance')])
+
+        runLoop = 1
+        countPasses = 1
+        countPassesDos = 1
+        resistanceShiftAmount = module.getModifiedItemAttr('resistanceShiftAmount')/100
+        while runLoop == 1:
+
+            for i in range(4):
+                # Update tuple with current values
+                attr = "armor%sDamageResonance" % damagePattern_tuple[i][0].capitalize()
+                damagePattern_tuple[i][3] = damagePattern_tuple[i][2] * damagePattern_tuple[i][1]
+                damagePattern_tuple[i][4] = module.getModifiedItemAttr(attr)
+                #logger.debug("Update values (Type|Resist|ModifiedDamage|AdaptiveResists): %s | %f | %f | %f", damagePattern_tuple[i][0], fit.ship.getModifiedItemAttr(attr), damagePattern_tuple[i][3], damagePattern_tuple[i][4])
+
+            # Sort the tuple by which resist took the most damage
+            damagePattern_tuple = sorted(damagePattern_tuple, key=operator.itemgetter(3,0))
+
+            if damagePattern.emAmount == damagePattern.thermalAmount == damagePattern.kineticAmount == damagePattern.explosiveAmount:
+                # If damage pattern is even across the board, we "reset" back to original resist values.
+                logger.debug("Setting adaptivearmorhardener resists to uniform profile.")
+                # Do nothing, because the RAH gets recalculated, so just don't change it
+                runLoop = 0
```

Comment:
>Why use the default profile for uniform damage instead of actually seeing how the RAH changes?

We could, but the primary issue is that there is no easy way to distinguish between the default damage profile and a damage profile that has uniform damage.

I want to be able to get back to the default, so that people who don't use the damage profiles get the behavior they have come to expect.  It also lets them see the difference in how the damage reduction looks when using RAH with defaults (old version) and the new reactive version.

If there was a clear indication which damage profile you had selected (and not just what the percentages are), then I'd be more inclined to do exactly this.

Until that happens though, it's very difficult to see at a glance which damage profile you have selected.